### PR TITLE
Remove ENABLE_WALLET from libbitcoin_server.a

### PR DESCRIPTION
**Pull request by jnewbery**
This removes the final instances of ENABLE_WALLET in libbitcoin_server and so completes #7965.

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Strong Concept ACK

**Review by ghost — APPROVED** [review]

**Comment by laanwj** [comment]
Yayyyy, good to see this finally happen
utACK 9a2b3d2e9d5584adb19633485b5398accc3704b7

**Comment by donaloconnor** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/14168/commits/9a2b3d2e9d5584adb19633485b5398accc3704b7 - looks much cleaner :+1:

**Comment by ken2812221** [comment]
utACK 9a2b3d2

File: src/wallet/init.cpp [hunk 1]
```diff
@@ -19,6 +19,9 @@
 class WalletInit : public WalletInitInterface {
 public:
 
+    //! Is the wallet component enabled
+    bool IsWalletEnabled() const override {return true;}
```
[the remark this answers is not in the document]
Comment (promag) [reply src/wallet/init.cpp]: My suggestion `HasWalletSupport()`.

[the remark this answers is not in the document]
Comment (jnewbery) [reply src/wallet/init.cpp]: Changed to `HasWalletSupport()`

**Comment by jnewbery** [comment]
Renamed function as suggested here: https://github.com/bitcoin/bitcoin/pull/14168#discussion_r216163230

**Comment by Empact** [comment]
utACK 7d038dc
nit: `HasWalletSupport` could be called `Enabled` for consistency with `ENABLE_WALLET`

**Comment by MarcoFalke** [comment]
utACK 7d038dcb41d93ea0b7401140998225adaab8c065

**Pull request merged**